### DISCUSSION TITLE
[Feature Request] Pagination for blog's ArticleList

### DISCUSSION THREAD

**Discussion by vmihailenco** [vuepress-theme-hope/vuepress-theme-hope#3171] [Q&A] [answered]
### Checklist

- [X] I searched existing issues and no one else requests similar feature.
- [X] I think that 25%+ users are positive towards this feature.

### Describe the feature

Large blogs that have many articles need some pagination or the index page becomes too large which causes issues for users & crawlers.

### Additional context

Doing pagination on the client is probably not that hard, but having SSR support would be nice.

**Comment by Mister-Hope** [comment]
Isn't there a pagination already?

What do you mean?

**Comment by vmihailenco** [comment]
I am using this [example](https://github.com/vuepress-theme-hope/vuepress-theme-hope/tree/main/demo/blog2) with the defalt vuepress theme: [blog](https://uptrace.dev/blog/). I would like to have a pagination at the bottom of the page so I don't have too many posts on a single page.

It could be that it is already supported, but I've checked with 21 posts and no pagination magically appears...

**Comment by Mister-Hope** [comment, marked as answer]
The blog plugin is designed to provide core blog features, and pagination is some kind of process of blog data at client side, which is not a part of core blogging feature. A pagination component is also something at viewer level

VuePress Theme Hope already supports this by adding a pagination component. If you want blog plugin used with other themes, it's their job to create such a component fitting their own style.

**Reply by vmihailenco** [reply]
>A pagination component is also something at viewer level

When I used vitepress it generated pages when the blog was compiled. So it can be done using SSR.

It also **should** be done using SSR, because otherwise you only harm user experience and make indexing your blog hard or even impossible.

If you check your blog, you will see that all your pages have the same size when downloaded as HTML:
- https://theme-hope.vuejs.press/demo/blog-home.html
- https://theme-hope.vuejs.press/demo/blog-home.html?page=2
- etc

So you might as well remove the pagination and let users scroll which is easier than using pagination.

It will also improve SEO, because currently crawlers that don't have JS can only index the first page. The 2nd page returns all the same HTML like the first one and it is only updated on the client.

**Reply by Mister-Hope** [reply]
That's why we have sitemap, you shouldn't rely on links in page to let crawlers get them, it's possible that you have a single page which is not referenced anywhere.

Also, you should not expect search engine crawlers to crawl any dynamic pages (for a static blog site, article pages are). The posts page should be places where actual contents are, and everything on post list should be in those pages as well.

Engines like google will not reindex pages same as the frequency you update your site. And once you update site, most search engines only reindex some of the existing pages, and add some of the new pages, a lot of existing pages shall wait for their next reindex schedule to update the index.

There could be chances where you add a lot of articles at one time (e.g. 10), and once you have 10 article per page, Google may index the new page /xxx/5 and mark it as a copy of /xxx/4, and it just starts a reindex schedule of 30 days for the page.

The best thing to do SEO is theme-hope does, provide a sitemap excluding all dynamic page. **All information can be found at post pages and they are all included in sitemap, and users won't happened to enter an article list page where the index content is outdated, and leave without finding any matched content.**
